lib/ast/node.rb

```diff
@@ -227,6 +227,21 @@ def inspect(indent=0)
     def to_ast
       self
     end
+    
+    # Produces a machine-readable s-expression.
+    #
+    # @return [Array]
+    def to_json
```

Comment:
This doesn't really return JSON though, it returns an array. Needs a better name.

Reply:
Yes, this needs a better name. Thoughts?

Reply:
I was thinking perhaps `to_a`? Could get rid of the `to_s` since I believe `JSON.generate` will handle symbols just fine.

Reply:
There is already .to_a, and removing any methods would break API.

Reply:
Ah, it looks like `to_a` is taken (just returns `children`). 

Hmm...

Reply:
I assume you'd rather have the actual `JSON.generate` call take place in `parser`, not here – `ast` should probably not `require 'json'`, I assume. 

Is that correct?

Reply:
Maybe `.to_plain`?

Reply:
Updated

Reply:
Don't like it.